File: src/org/scalaconsole/fxui/manual/ManualController.java
package org.scalaconsole.fxui.manual;

import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.scene.control.TextField;
import javafx.scene.input.KeyEvent;
import javafx.stage.Stage;

public class ManualController {
    ManualDelegate delegate;

    @FXML
    TextField groupId;

    @FXML
    TextField artifactId;

    @FXML
    TextField version;

    @FXML
    void initialize() {
        delegate = new ManualDelegate();
    }

    @FXML
    void onOK(ActionEvent e) {
        ManualStage window = (ManualStage) groupId.getScene().getWindow();
        window.onOK();
        window.close();
    }

    @FXML
    void onCancel(ActionEvent e) {
        Stage manualStage = (Stage) groupId.getScene().getWindow();
        manualStage.close();
    }

    @FXML
    void onKeyUp(KeyEvent e) {
        switch (e.getCode()) {
            case ENTER:
                if (!(groupId.getText().isEmpty() || artifactId.getText().isEmpty() || version.getText().isEmpty()))
                    onOK(null);
                break;
            case ESCAPE:
                onCancel(null);
                break;
        }
    }

}

File: src/org/scalaconsole/fxui/manual/ManualController.scala
package org.scalaconsole.fxui.manual
import javafx.stage.Stage
import javafx.fxml.FXML
import javafx.event.ActionEvent
import javafx.scene.input.{KeyCode, KeyEvent}
import KeyCode._

trait ManualController { self: ManualStage =>
  @FXML  def onOK(e: ActionEvent) {
    try {
      val artifactString = s"${groupId.getText.trim}:${artifactId.getText.trim}:${version.getText.trim}"
      mainStage.addArtifacts(artifactString :: Nil)
    } catch {
      case e: Throwable => e.printStackTrace()
    }
    close()
  }

  @FXML  def onCancel(e: ActionEvent) {
    close()
  }

  @FXML  def onKeyUp(e: KeyEvent) {
    e.getCode match {
      case ENTER =>
        if (!(groupId.getText.isEmpty || artifactId.getText.isEmpty || version.getText.isEmpty)) onOK(null)
      case ESCAPE =>
        close()
      case _ =>
    }
  }
}
